src/main/java/com/microsoftopentechnologies/azure/AzureCloud.java

```diff
@@ -388,31 +423,31 @@ public void doProvision(StaplerRequest req, StaplerResponse rsp, @QueryParameter
 		
 		LOGGER.severe("Azure Cloud: doProvision: creating slave ");
 		Computer.threadPoolForRemoting.submit(new Callable<Node>() {
-			
+
 			public Node call() throws Exception {
 				@SuppressWarnings("deprecation")
 				AzureSlave slave = slaveTemplate.provisionSlave(new StreamTaskListener(System.out));
 				// Get virtual machine properties
-				LOGGER.info("Azure Cloud: provision: Getting virtual machine properties for slave "+slave.getNodeName() 
-						+ " with OS "+slave.getOsType());
+				LOGGER.info("Azure Cloud: provision: Getting virtual machine properties for slave " + slave.getNodeName()
+						+ " with OS " + slave.getOsType());
 				slaveTemplate.setVirtualMachineDetails(slave);
-				
+
 				if (slave.getSlaveLaunchMethod().equalsIgnoreCase("SSH")) {
-					 slaveTemplate.waitForReadyRole(slave);
-					 LOGGER.info("Azure Cloud: provision: Waiting for ssh server to come up");
-					 Thread.sleep(2 * 60 * 1000);
-					 LOGGER.info("Azure Cloud: provision: ssh server may be up by this time");
-					 LOGGER.info("Azure Cloud: provision: Adding slave to azure nodes ");
-					 Hudson.getInstance().addNode(slave);
-					 slave.toComputer().connect(false).get();
-				 } else if (slave.getSlaveLaunchMethod().equalsIgnoreCase("JNLP")) {
-					 LOGGER.info("Azure Cloud: provision: Checking for slave status");
-					 slaveTemplate.waitForReadyRole(slave);
-					 Hudson.getInstance().addNode(slave);
-					 
-					 // Wait until node is online
-					 waitUntilOnline(slave);
-				 }
+					slaveTemplate.waitForReadyRole(slave);
+					LOGGER.info("Azure Cloud: provision: Waiting for ssh server to come up");
+					Thread.sleep(2 * 60 * 1000);
+					LOGGER.info("Azure Cloud: provision: ssh server may be up by this time");
+					LOGGER.info("Azure Cloud: provision: Adding slave to azure nodes ");
+					Hudson.getInstance().addNode(slave);
+					slave.toComputer().connect(false).get();
```

Comment:
toComputer() may return null if the node has been reconfigured and we still use an old instance